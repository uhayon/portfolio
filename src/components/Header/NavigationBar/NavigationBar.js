import React from 'react';
import PropTypes from 'prop-types';

import LanguageSwitcher from './LanguageSwitcher/LanguageSwitcher';
import ToggleNavigationBarButton from '../ToggleNavigationBarButton/ToggleNavigationBarButton';
import NavigationButtonsContainer from './NavigationButtonsContainer/NavigationButtonsContainer';

import styles from './NavigationBar.module.scss';

const NavigationBar = ({ selectedLanguage, handleLanguageSwitch, isExpanded, isSideBar, toggleSidebarExpansion, buttons }) => {
  return (
    <>
      {
        isSideBar ?
        <div 
        className={`${styles.sidebarContainer} ${isExpanded ? styles.sidebarExpanded : ''}`}
        onClick={toggleSidebarExpansion}>
        </div> :
        null
      }
      <nav 
        onClick={(event) => event.stopPropagation()}
        className={`${isSideBar ? styles.sidebar : styles.navbar} ${isSideBar && isExpanded ? styles.sidebarOpen : ''}`}>
        <div className={styles.closeButtonContainer}>
          <ToggleNavigationBarButton
            isMobileScreen={isSideBar}
            handleNavButtonPress={toggleSidebarExpansion} />
        </div>
        <NavigationButtonsContainer selectedLanguage={selectedLanguage} buttons={buttons} isSideBar={isSideBar} />
        <LanguageSwitcher selectedLanguage={selectedLanguage} handleLanguageSwitch={handleLanguageSwitch} />
      </nav>
    </>
  )
};

NavigationBar.propTypes = {
  selectedLanguage: PropTypes.string.isRequired,
  handleLanguageSwitch: PropTypes.func.isRequired,
  isExpanded: PropTypes.bool,
  isSideBar: PropTypes.bool,
  toggleSidebarExpansion: PropTypes.func.isRequired,
  buttons: PropTypes.array.isRequired
}

export default NavigationBar;