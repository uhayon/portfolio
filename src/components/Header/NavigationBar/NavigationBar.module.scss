@import '../../../globalStyles/colors.scss';

.sidebarContainer {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background-color: rgba(0, 0, 0, .75);
    z-index: 1001;
    display: none;
    opacity: 0;
    transition: opacity 0.5s linear;

    &.sidebarExpanded {
        display: block;
        opacity: 1;
        min-height: 100vh;
        min-width: 100vw;
    }
}

.sidebar {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 0;
    padding: 0;
    transition: width 0.3s ease-in;
    position: absolute;
    background-color: $gray-dark;
    top: 0;
    right: 0;
    z-index: 1002;
    min-height: 100vh;
    overflow: auto;
    padding: 1rem 0;
    > * {
        opacity: 0;
    }

    &.sidebarOpen {
        width: 100%;
        padding: 1rem .5rem;

        > * {
            opacity: 1;
        }

        .closeButtonContainer {
            display: flex;
            justify-content: flex-end;
            width: 100%;
        }
    }

    @media screen and (min-width: 500px) {
        &.sidebarOpen {
            width: 250px;
        }
    }
}

.navbar {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 100%;
}