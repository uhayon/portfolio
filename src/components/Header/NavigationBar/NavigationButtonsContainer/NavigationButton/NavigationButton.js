import React from 'react';

import styles from './NavigationButton.module.scss';

const NavigationButton = ({ selectedLanguage, buttonOptions }) => (
  <li className={styles.navigationButton}>
    <a href={`#${buttonOptions.sectionId}`} className={styles.link}>{buttonOptions.title[selectedLanguage]}</a>
  </li>
);

export default NavigationButton;