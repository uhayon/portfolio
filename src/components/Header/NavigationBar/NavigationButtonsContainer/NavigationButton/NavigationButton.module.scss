@import '../../../../../globalStyles/colors.scss';

.navigationButton {
    padding: 1rem 0.5rem;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
    box-sizing: border-box;

    &:hover {
        .link {
            color: $red;
        }
    }

    .link {
        text-decoration: none;
        font-weight: bold;
        font-size: 1.2rem;
        color: $white;
        transition: color 0.2s ease-in;
    }
}