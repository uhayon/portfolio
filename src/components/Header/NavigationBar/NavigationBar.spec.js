import React from 'React';
import { shallow, mount } from 'enzyme';

import NavigationBar from './NavigationBar';
import NavigationButton from './NavigationButtonsContainer/NavigationButton/NavigationButton';
import navigationButtons from '../NavigationBarButtons.json';

describe('Mounting NavigationBar', () => {
  let component;
  beforeEach(() => component = <NavigationBar selectedLanguage='EN' handleLanguageSwitch={jest.fn()} isExpanded={false} isSideBar={false} toggleSidebarExpansion={jest.fn()} buttons={navigationButtons} />);

  it('Must match snapshot', () => {
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();    
  })

  it('Must match titles by selected language', () => {
    const wrapper = mount(component)
    expect(wrapper.find(NavigationButton).length).toEqual(navigationButtons.length); 

    wrapper.find(NavigationButton).forEach((navButton, i) => {
      expect(navButton.find('a').text()).toEqual(navigationButtons[i].title.EN);
    });

    wrapper.setProps({ selectedLanguage: 'ES' });
    wrapper.find(NavigationButton).forEach((navButton, i) => {
      expect(navButton.find('a').text()).toEqual(navigationButtons[i].title.ES);
    });
  })
});