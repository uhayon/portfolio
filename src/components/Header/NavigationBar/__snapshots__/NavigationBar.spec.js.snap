// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounting NavigationBar Must match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationBar
    buttons={
      Array [
        Object {
          "sectionId": "section-skills",
          "title": Object {
            "EN": "Skills",
            "ES": "Habilidades",
          },
        },
        Object {
          "sectionId": "section-projects",
          "title": Object {
            "EN": "Projects",
            "ES": "Proyectos",
          },
        },
        Object {
          "sectionId": "section-contact",
          "title": Object {
            "EN": "Contact Me",
            "ES": "Contactame",
          },
        },
      ]
    }
    handleLanguageSwitch={[MockFunction]}
    isExpanded={false}
    isSideBar={false}
    selectedLanguage="EN"
    toggleSidebarExpansion={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        <nav
          className="navbar "
          onClick={[Function]}
        >
          <div
            className="closeButtonContainer"
          >
            <ToggleNavigationBarButton
              handleNavButtonPress={[MockFunction]}
              isMobileScreen={false}
            />
          </div>
          <NavigationButtonsContainer
            buttons={
              Array [
                Object {
                  "sectionId": "section-skills",
                  "title": Object {
                    "EN": "Skills",
                    "ES": "Habilidades",
                  },
                },
                Object {
                  "sectionId": "section-projects",
                  "title": Object {
                    "EN": "Projects",
                    "ES": "Proyectos",
                  },
                },
                Object {
                  "sectionId": "section-contact",
                  "title": Object {
                    "EN": "Contact Me",
                    "ES": "Contactame",
                  },
                },
              ]
            }
            isSideBar={false}
            selectedLanguage="EN"
          />
          <LanguageSwitcher
            handleLanguageSwitch={[MockFunction]}
            selectedLanguage="EN"
          />
        </nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="closeButtonContainer"
            >
              <ToggleNavigationBarButton
                handleNavButtonPress={[MockFunction]}
                isMobileScreen={false}
              />
            </div>,
            <NavigationButtonsContainer
              buttons={
                Array [
                  Object {
                    "sectionId": "section-skills",
                    "title": Object {
                      "EN": "Skills",
                      "ES": "Habilidades",
                    },
                  },
                  Object {
                    "sectionId": "section-projects",
                    "title": Object {
                      "EN": "Projects",
                      "ES": "Proyectos",
                    },
                  },
                  Object {
                    "sectionId": "section-contact",
                    "title": Object {
                      "EN": "Contact Me",
                      "ES": "Contactame",
                    },
                  },
                ]
              }
              isSideBar={false}
              selectedLanguage="EN"
            />,
            <LanguageSwitcher
              handleLanguageSwitch={[MockFunction]}
              selectedLanguage="EN"
            />,
          ],
          "className": "navbar ",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ToggleNavigationBarButton
                handleNavButtonPress={[MockFunction]}
                isMobileScreen={false}
              />,
              "className": "closeButtonContainer",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleNavButtonPress": [MockFunction],
                "isMobileScreen": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttons": Array [
                Object {
                  "sectionId": "section-skills",
                  "title": Object {
                    "EN": "Skills",
                    "ES": "Habilidades",
                  },
                },
                Object {
                  "sectionId": "section-projects",
                  "title": Object {
                    "EN": "Projects",
                    "ES": "Proyectos",
                  },
                },
                Object {
                  "sectionId": "section-contact",
                  "title": Object {
                    "EN": "Contact Me",
                    "ES": "Contactame",
                  },
                },
              ],
              "isSideBar": false,
              "selectedLanguage": "EN",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleLanguageSwitch": [MockFunction],
              "selectedLanguage": "EN",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "nav",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          <nav
            className="navbar "
            onClick={[Function]}
          >
            <div
              className="closeButtonContainer"
            >
              <ToggleNavigationBarButton
                handleNavButtonPress={[MockFunction]}
                isMobileScreen={false}
              />
            </div>
            <NavigationButtonsContainer
              buttons={
                Array [
                  Object {
                    "sectionId": "section-skills",
                    "title": Object {
                      "EN": "Skills",
                      "ES": "Habilidades",
                    },
                  },
                  Object {
                    "sectionId": "section-projects",
                    "title": Object {
                      "EN": "Projects",
                      "ES": "Proyectos",
                    },
                  },
                  Object {
                    "sectionId": "section-contact",
                    "title": Object {
                      "EN": "Contact Me",
                      "ES": "Contactame",
                    },
                  },
                ]
              }
              isSideBar={false}
              selectedLanguage="EN"
            />
            <LanguageSwitcher
              handleLanguageSwitch={[MockFunction]}
              selectedLanguage="EN"
            />
          </nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="closeButtonContainer"
              >
                <ToggleNavigationBarButton
                  handleNavButtonPress={[MockFunction]}
                  isMobileScreen={false}
                />
              </div>,
              <NavigationButtonsContainer
                buttons={
                  Array [
                    Object {
                      "sectionId": "section-skills",
                      "title": Object {
                        "EN": "Skills",
                        "ES": "Habilidades",
                      },
                    },
                    Object {
                      "sectionId": "section-projects",
                      "title": Object {
                        "EN": "Projects",
                        "ES": "Proyectos",
                      },
                    },
                    Object {
                      "sectionId": "section-contact",
                      "title": Object {
                        "EN": "Contact Me",
                        "ES": "Contactame",
                      },
                    },
                  ]
                }
                isSideBar={false}
                selectedLanguage="EN"
              />,
              <LanguageSwitcher
                handleLanguageSwitch={[MockFunction]}
                selectedLanguage="EN"
              />,
            ],
            "className": "navbar ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ToggleNavigationBarButton
                  handleNavButtonPress={[MockFunction]}
                  isMobileScreen={false}
                />,
                "className": "closeButtonContainer",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleNavButtonPress": [MockFunction],
                  "isMobileScreen": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "buttons": Array [
                  Object {
                    "sectionId": "section-skills",
                    "title": Object {
                      "EN": "Skills",
                      "ES": "Habilidades",
                    },
                  },
                  Object {
                    "sectionId": "section-projects",
                    "title": Object {
                      "EN": "Projects",
                      "ES": "Proyectos",
                    },
                  },
                  Object {
                    "sectionId": "section-contact",
                    "title": Object {
                      "EN": "Contact Me",
                      "ES": "Contactame",
                    },
                  },
                ],
                "isSideBar": false,
                "selectedLanguage": "EN",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleLanguageSwitch": [MockFunction],
                "selectedLanguage": "EN",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
